Detectors/TPC/simulation/include/TPCSimulation/DigitPad.h

```diff
@@ -53,41 +60,35 @@ class DigitPad{
     /// \param pad pad ID
     void fillOutputContainer(TClonesArray *output, int cru, int timeBin, int row, int pad, float commonMode = 0);
 
-    /// The MC labels are sorted by occurrence such that the event/track combination with the largest number of occurrences is first
-    /// This is then dumped into a std::vector and attached to the digits
-    /// \todo Find out how many different event/track combinations are relevant
-    /// \param std::vector containing the sorted MCLabels
-    void processMClabels(std::vector<long> &sortedMCLabels);
-    
   private:
     float                  mChargePad;   ///< Total accumulated charge on that pad for a given time bin
     unsigned char          mPad;         ///< Pad of the ADC value
-    std::map<long, int>    mMCID;        ///< Map containing the MC labels (key) and the according number of occurrence (value)
+    FairMultiLinkedData    mMCLinks;     ///< MC links
 };
 
 inline
 DigitPad::DigitPad(int pad)
   : mChargePad(0.)
   , mPad(pad)
-  , mMCID()
+  , mMCLinks()
 {}
 
 inline 
-void DigitPad::setDigit(int eventID, int trackID, float charge)
+void DigitPad::setDigit(size_t hitID, float charge)
 {
   /// the MC ID is encoded such that we can have 999,999 tracks
   /// numbers larger than 1000000 correspond to the event ID
   /// i.e. 12000010 corresponds to event 12 with track ID 10
   /// \todo Faster would be a bit shift
-  ++mMCID[(eventID)*1000000 + trackID];
+  mMCLinks.AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), "TPCPoint", hitID));
```

Comment:
is `"TPCPoints"` just a descriptive information? I suggest adding a constant for this